jujugui/static/gui/src/app/app.js

```diff
@@ -527,24 +525,16 @@ YUI.add('juju-gui', function(Y) {
       // Store the initial model UUID.
       const modelUUID = this._getModelUUID();
       this.set('modelUUID', modelUUID);
-      // If the user closed the GUI when they were on a different env than
-      // their default then it would show them the login screen. This sets
-      // the credentials to the environment that they are logging into
-      // initially.
-      var user = modelAPI.get('user');
-      var password = modelAPI.get('password');
+      var user = null;
+      var password = null;
       var macaroons = null;
-      if (!user || !password) {
-        // No user and password credentials provided in config: proceed with
-        // usual credentials handling.
-        var credentials = modelAPI.getCredentials();
-        if (credentials.areAvailable) {
-          user = credentials.user;
-          password = credentials.password;
-          macaroons = credentials.macaroons;
-        }
+      var credentials = this.user.controller;
+      if (credentials.areAvailable) {
+        user = credentials.user;
+        password = credentials.password;
+        macaroons = credentials.macaroons;
       }
-      modelAPI.setCredentials({ user, password, macaroons });
+      this.user.controller = { user, password, macaroons };
```

Comment:
In the next pass we can also standarize on null or '' representing no data for user/pass fields and then make sure the controller always returns whichever and everything else accepts that. I lean towards null myself.